### DISCUSSION TITLE
What is the intended modal editor method  that negates the need to navigate in insert mode?

### DISCUSSION THREAD

**Discussion by Ryex** [helix-editor/helix#3790] [Q&A] [answered]
So, I'll be honest the recent removal of navigation bindings in insert mode *really* tripped me up. And on reading the pull request discussion I guess I get the motivation.

But this just leaves me scratching my head because there ARE IMO legitimate reasons to move a character or two while doing an insert before dropping back to normal mode.
IE. a minor typo where you don't want to backspace 4-5 characters and then retype?

Or my current major annoyance before re-adding the navigation keybindings. 

moving out of auto inserted matching quote or brace pairs, or to the end of a line to add a comma when working with data structures. It makes absolutely no sense to me to exit insert mode arrow over a single character and reenter insert mode  just because I've reach then end of the `()|[]|{}|''|""` pair I'm typing. Sometimes I'll need to move up to three characters to exit  a `"})` line up of ending pairs to keep going.

am I missing something? is there a better way to handle these cases in model mode I'm ignorant of?

**Comment by Omnikar** [comment]
You can exit a pair by typing the closing character e.g.
`(abc|)`
Type `)`.
`(abc)|`

**Reply by Ryex** [reply]
I didn't remember this at the time but I have used this feature without realizing it a ton and should have noticed. that said it still feels wrong. thanks for reminding me though.

**Comment by the-mikedavis** [comment, marked as answer]
Re-introducing arrow-key bindings into your config is perfectly valid IMO. As you say, being able to move the cursor a tiny amount without exiting normal-mode and re-entering insert-mode can be useful.

Tabout (https://github.com/helix-editor/helix/issues/1587) would be the ideal way to jump out of closing pairs. I have a PR for that which gets close but needs refinement: https://github.com/helix-editor/helix/pull/3350

In general though, motions/navigation should be done in normal-mode. <kbd>Esc</kbd><kbd>l</kbd><kbd>i</kbd> is a longer sequence than pressing the right arrow key but you don't need to move your hand to the arrow keys, and you get an undo checkpoint from switching to normal mode. The more you need to move, the less costly the switch to normal mode becomes. And if you remap caps-lock as Esc - which I would highly recommend - then the entire motion can be done without leaving touch-typing position.

**Reply by Ryex** [reply]
This is the kind of feature I was thinking of. good to know It has so advocators. Looking forward to it.
